hovercontrol/src/main.rs

```diff
@@ -1,265 +1,48 @@
-                self.send_target(Side::Right)?;
-            }
-            EventType::ButtonPressed(Button::RightTrigger2, _code) => {
-                self.centre_right -= CENTRE_STEP;
-                self.send_target(Side::Right)?;
-            }
-            EventType::ButtonPressed(Button::LeftThumb, _code) => {
-                self.centre_left = 0;
-                self.hoverkite.send_command(Side::Left, Command::Recenter)?;
-            }
-            EventType::ButtonPressed(Button::RightThumb, _code) => {
-                self.centre_right = 0;
-                self.hoverkite
-                    .send_command(Side::Right, Command::Recenter)?;
-            }
-            EventType::ButtonPressed(Button::South, _code) => {
-                self.hoverkite
-                    .send_command(Side::Left, Command::ReportBattery)?;
-                self.hoverkite
-                    .send_command(Side::Right, Command::ReportBattery)?;
-            }
-            EventType::ButtonPressed(Button::East, _code) => {
-                self.hoverkite
-                    .send_command(Side::Left, Command::RemoveTarget)?;
-                self.hoverkite
-                    .send_command(Side::Right, Command::RemoveTarget)?;
-            }
-            EventType::ButtonPressed(Button::West, _code) => {
-                if self.spring_constant > SPRING_CONSTANT_STEP {
-                    self.spring_constant -= SPRING_CONSTANT_STEP;
-                    self.send_spring_constant()?;
-                }
-            }
-            EventType::ButtonPressed(Button::North, _code) => {
-                if self.spring_constant < MAX_SPRING_CONSTANT {
-                    self.spring_constant += SPRING_CONSTANT_STEP;
-                    self.send_spring_constant()?;
-                }
-            }
-            EventType::ButtonPressed(Button::Mode, _code) => {
-                // Power off
-                self.hoverkite.send_command(Side::Left, Command::PowerOff)?;
-                self.hoverkite
-                    .send_command(Side::Right, Command::PowerOff)?;
-            }
-            EventType::ButtonPressed(button, code) => {
-                println!("Button {:?} pressed: {:?}", button, code);
-            }
-            _ => {}
-        }
-        Ok(())
-    }
-
-    pub fn send_max_speed(&mut self) -> Result<(), Report> {
-        // Invert left
-        self.hoverkite
-            .set_max_speed(Side::Left, self.max_speed.invert())?;
-        self.hoverkite.set_max_speed(Side::Right, self.max_speed)?;
-        Ok(())
-    }
+    let gilrs = Gilrs::new().unwrap();
 
-    fn send_spring_constant(&mut self) -> Result<(), Report> {
-        self.hoverkite
-            .set_spring_constant(self.spring_constant)
-            .wrap_err("Failed to set spring constant")
-    }
+    let runtime = Runtime::new()?;
+    let handle = runtime.handle();
+    let homie = Homie::connect_and_start(handle, config.mqtt)?;
```

Comment:
dump the runtime creation in the constructor of your Homie object and hold onto it as a field in Homie?

Reply:
Done.